lib/resolve-package-path.js

```diff
@@ -0,0 +1,105 @@
+// TODO: consider upstreaming to stefanpenner/resolve-package-path
+
+const fs = require('fs');
+const path = require('path');
+const pathRoot = require('path-root');
+
+class Cache {
```

Comment:
@stefanpenner this whole file is the thing that we might want to replace with another function in `resolve-package-path` whose semantics are "find my project's package path" for a directory within a project.